usr/share/rear/lib/global-functions.sh

```diff
@@ -558,65 +619,135 @@ mount_url() {
+    # Therefore it also determines if exit task and mountpoint handling is required and returns early if not.
+    # The actual umount job is performed inside perform_umount_url().
+    # We do not request lazy umount here because we want umount errors to be reliably reported.
+    perform_umount_url $url $mountpoint || Error "Unmounting '$mountpoint' failed."
 
-            RemoveExitTask "umount -f $v '$mountpoint' >&2"
-            return 0
+    RemoveExitTask "perform_umount_url '$url' '$mountpoint' y"
+
+    remove_temporary_mountpoint '$mountpoint' && RemoveExitTask "remove_temporary_mountpoint '$mountpoint'"
+    return 0
+}
+
+### Unmount url $1 at mountpoint $2 [ lazily if $3 is set and normal unmount fails ]
+function perform_umount_url() {
+    local url=$1
+    local mountpoint=$2
+    local lazy=${3:-}
+
+    case $(url_scheme $url) in
+        (sshfs)
+            # does ftpfs need this special case as well?
+            fusermount -u ${lazy:+'-z'} $mountpoint
+            ;;
+        (davfs)
+            umount_davfs $mountpoint $lazy
+            ;;
+        (var)
+            local var
+            var=$(url_host $url)
+            Log "Unmounting with '${!var} $mountpoint'"
+            # lazy unmount not supported with custom umount command
+            ${!var} $mountpoint
             ;;
+        (*)
+            # usual umount command
+            umount_mountpoint $mountpoint $lazy
     esac
+    # The switch above must be the last statement in this function and the umount commands must be
+    # the last commands (or part of) in each branch. This ensures proper exit code propagation
+    # to the caller even when set -e is used.
+}
 
-    umount_mountpoint $mountpoint
-    StopIfError "Unmounting '$mountpoint' failed."
-
-    RemoveExitTask "umount -f $v '$mountpoint' >&2"
-    return 0
+### Helper which unmounts davfs mountpoint $1 and cleans up the cache,
+### performing lazy unmount if $2 is nonempty and normal unmount fails.
+function umount_davfs() {
+    local mountpoint=$1
+    local lazy="${2:-}"
+    if umount_mountpoint $mountpoint ; then
+        # Wait for 3 sek. then remove the cache-dir /var/cache/davfs
+        sleep 30
+        # ToDo: put in here the cache-dir from /etc/davfs2/davfs.conf
+        # and delete only the just used cache
+        #rm -rf /var/cache/davfs2/*<mountpoint-hash>*
+        rm -rf /var/cache/davfs2/*outputfs*
+    else
+        local retval=$?
+
+        if test $lazy ; then
+            # try again to unmount lazily and this time do not delete the cache, it is still in use.
+            LogPrintError "davfs cache /var/cache/davfs2/*outputfs* needs to be cleaned up manually after the lazy unmount finishes"
+            umount_mountpoint_lazy $mountpoint
+        else
+            # propagate errors from umount
+            return $retval
+        fi
+    fi
 }
 
-### Unmount mountpoint $1
-umount_mountpoint() {
+### Unmount mountpoint $1 [ lazily if $2 is nonempty ]
```

Comment:
I would prefer a more explicit syntax for the `$2` "lazy" parameter
because currently `perform_umount_url '$url' '$mountpoint' yes`
does not tell what that `yes` means so the code is less obvious
but even worse `perform_umount_url '$url' '$mountpoint' no`
is same as `perform_umount_url '$url' '$mountpoint' yes`
and that is really ugly - but matches well-known ugly C syntax
style like `if(!strcmp(str1, str2)) printf("strings are equal");`

More explicit syntax could be that `$2` must be either the exact word `lazy`
for lazy umount or empty for non-lazy umount so there might be even
a `BugError` in umount_mountpoint() if it was called with wrong `$2` value.

Reply:
Done via https://github.com/rear/rear/pull/2625/commits/6ddce251db788f04935c8fcb3ba74a826405b042
@pcahyna thank you!